src/api/app/models/workflow/step/branch_package_step.rb

```diff
@@ -15,10 +15,12 @@ def target_project_base_name
     step_instructions[:target_project]
   end
 
-  def branch_package(workflow_filters = {})
+  def branch_package(workflow_filters = {})    
     create_branched_package if scm_webhook.new_pull_request? || (scm_webhook.updated_pull_request? && target_package.blank?) || scm_webhook.push_event? || scm_webhook.tag_push_event?
 
-    add_branch_request_file(package: target_package)
+    add_branch_request_file(package: target_package) unless scm_synced?
```

Comment:
Combining this line with the next one in a _if/else_ statement would make this easier to read. The condition would also not be checked twice, only once.